src/ompl/base/samplers/informed/src/PathLengthDirectInfSampler.cpp

```diff
@@ -117,19 +117,21 @@ namespace ompl
             {
                 // Check that it is SE2 or SE3
                 if (InformedSampler::space_->getType() == STATE_SPACE_SE2 ||
-                    InformedSampler::space_->getType() == STATE_SPACE_SE3)
+                    InformedSampler::space_->getType() == STATE_SPACE_SE3 || 
+                    InformedSampler::space_->getType() == 98)
                 {
                     // Variable:
                     // An ease of use upcasted pointer to the space as a compound space
                     const CompoundStateSpace *compoundSpace = InformedSampler::space_->as<CompoundStateSpace>();
 
-                    // Sanity check
+                    // Sanity check - Removed
+                    /*
                     if (compoundSpace->getSubspaceCount() != 2u)
                     {
                         // Pout
                         throw Exception("The provided compound StateSpace is SE(2) or SE(3) but does not have exactly "
                                         "2 subspaces.");
-                    }
+                    }*/
```

Comment:
yes; this does not work with our compound space as we are combining SO2 with real vector spaces in a way not covered by this check.

Reply:
can we preserve the existing check and have it detect our unsupported new case and let it pass? we never want to remove any existing functionality from ompl because i wouldn't expect that to be accepted upstream.

Reply:
i added an if statement to check the state space we're in and skip if we're in the new one